src/auth/dto/authcreds.dto.ts

```diff
@@ -0,0 +1,15 @@
+import { IsString, IsNotEmpty, IsEmail } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class AuthCredsDto {
+  @IsString()
+  @ApiProperty({ description: 'The email of the user' })
+  @IsEmail()
+  @IsNotEmpty()
+  email: string;
+
+  @IsString()
```

Comment:
PullProofAI

The log message in the `getHello` method has been updated.
```typescript
-    console.log('Returning message');
+    console.log('Returning hello message');
```
This change improves clarity by providing a more descriptive log message.

### Changes in `AuthController` and related classes

---new_hunk---
```typescript
1: import { AuthService } from './auth.service';
2: 
3: @Controller('auth')
4: export class AuthController {
5:   constructor(private authService: AuthService) {}
6: 
7:   @Get('github/callback')
8:   githubCallback(@Req() req, @Res() res) {
9:     // GitHub authentication callback logic
10:   }
11: 
12:   @Post('logout')
13:   logout(@Req() req, @Res() res) {
14:     // User logout logic
15:   }
16: }
```

---old_hunk---
```typescript

```



<!-- This is an auto-generated comment by PullProofAI -->